### PR TITLE
fix resources in docker 1.7

### PR DESCRIPTION
**Pull request by vieux**
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

### CODE REVIEW

**Comment by vieux** [comment]
@abronan @aluzzardi if you see a better way to do `consolidateResourceFields` please tell me

**Comment by abronan** [comment]
Not really :-/ LGTM

File: cluster/config.go [hunk 1]
```diff
@@ -24,6 +24,40 @@ func parseEnv(e string) (bool, string, string) {
 	return false, "", ""
 }
 
+func consolidateResourceFields(c *dockerclient.ContainerConfig) {
```
Comment (aluzzardi) [inline cluster/config.go]:
```
// FIXME: Temporary fix to handle forward/backward compatibility between Docker <1.6 and >=1.7
// ContainerConfig should be handling converting to/from different docker versions
```

**Comment by aluzzardi** [comment]
The better way would be to handle converting one way using `BuildContainerConfig(<original docker config>, <api version>)` and the other way using `config.Serialize(<api version>)`, or something like that.

In the meantime, this does the trick but please add a FIXME as suggested.

LGTM

**Comment by vieux** [comment]
@aluzzardi done I also added a test, so the coverage doesn't decrease

**Pull request merged**